parlai/tasks/wizard_of_wikipedia/agents.py

```diff
@@ -303,7 +303,10 @@ def get(self, episode_idx, entry_idx=0):
             labels = [wizard_entry['text']]
         else:
             title, sentence = _get_chosen_title_and_sent(wizard_entry, knowledge_dict)
-            labels = ['{} {}'.format(title, sentence)]
+            if self.knowledge_separator and title != TOKEN_NOCHOSEN:
+                labels = ['{} {} {}'.format(title, TOKEN_KNOWLEDGE, sentence)]
+            else:
+                labels = ['{} {}'.format(title, sentence)]
 
         # finally, get label_candidates
         label_cands = [TOKEN_NOCHOSEN + ' ' + TOKEN_NOCHOSEN]
```

Comment:
Shouldn't you be checking for `self.knowledge_separator` here as well?

Reply:
i decided to not add the knowledge separator if the label is `TOKEN_NOCHOSEN`

Reply:
Ah, I just noticed the additional `title != TOKEN_NOCHOSEN` condition you added in line 306. Your decision makes sense now.

Just a minor note: if you can change the new line 312 to `label_cands = ["{} {}".format(TOKEN_NOCHOSEN, TOKEN_NOCHOSEN)]`, that would make it consistent with the rest of the code.